# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositableForwarder.h
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/ipc/CompositableForwarder.h
--- a/gfx/layers/ipc/CompositableForwarder.h	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/ipc/CompositableForwarder.h	Fri Mar 22 18:28:42 2019 +0000
@@ -81,9 +81,15 @@
    * TextureClient passed in parameter.
    * When the TextureClient has TEXTURE_DEALLOCATE_CLIENT flag,
    * the transaction becomes synchronous.
+   *
+   * aRenderRoot can be ignored if not using WebRender - since webrender
+   * splits the chrome and content areas into different documents which are
+   * updated separately, we need to know which command buffer to route this
+   * into.
    */
-  virtual void RemoveTextureFromCompositable(CompositableClient* aCompositable,
-                                             TextureClient* aTexture) = 0;
+  virtual void RemoveTextureFromCompositable(
+      CompositableClient* aCompositable, TextureClient* aTexture,
+      const Maybe<wr::RenderRoot>& aRenderRoot) = 0;
 
   struct TimedTextureClient {
     TimedTextureClient()
@@ -98,9 +104,15 @@
   /**
    * Tell the CompositableHost on the compositor side what textures to use for
    * the next composition.
+   *
+   * aRenderRoot can be ignored if not using WebRender - since webrender
+   * splits the chrome and content areas into different documents which are
+   * updated separately, we need to know which command buffer to route this
+   * into.
    */
   virtual void UseTextures(CompositableClient* aCompositable,
-                           const nsTArray<TimedTextureClient>& aTextures) = 0;
+                           const nsTArray<TimedTextureClient>& aTextures,
+                           const Maybe<wr::RenderRoot>& aRenderRoot) = 0;
   virtual void UseComponentAlphaTextures(CompositableClient* aCompositable,
                                          TextureClient* aClientOnBlack,
                                          TextureClient* aClientOnWhite) = 0;